Unable to compile on Windows

Hello,

I tried to compile 7.1.5 x64 TS VC14 with the master branch of pthreads
headers for pthread:
ftp://sourceware.org/pub/pthreads-win32/dll-latest/include/pthread.h

configure --disable-all --enable-cli --with-pthreads

"cl.exe" /D _USRDLL /D PHP7DLLTS_EXPORTS /D PHP_EXPORTS /D LIBZEND_EXPORTS /D TSRM_EXPORTS /D SAPI_EXPORTS /D WINVER=0x0600 /I "ext\pthreads" /DHAVE_PTHREAD_H=1 /DHAVE_SCHED_H=1 /DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 /Iext\pthreads /nologo /I . /I main /I Zend /I TSRM /I ext /D _WINDOWS /D ZEND_WIN32=1 /D PHP_WIN32=1 /D WIN32 /D _MBCS /W3 /FD /wd4996 /Zc:inline /MP /LD /MD /W3 /Ox /D NDebug /D NDEBUG /D ZEND_WIN32_FORCE_INLINE /GF /D ZEND_DEBUG=0 /D ZTS=1 /I "C:\php-sdk\phpdev\vc14\x64\deps\include" /D FD_SETSIZE=256 /FoC:\php-sdk\phpdev\vc14\x64\php-7.1.5-src\x64\Release_TS\ext\pthreads\ /FpC:\php-sdk\phpdev\vc14\x64\php-7.1.5-src\x64\Release_TS\ext\pthreads\ /FRC:\php-sdk\phpdev\vc14\x64\php-7.1.5-src\x64\Release_TS\ext\pthreads\ /FdC:\php-sdk\phpdev\vc14\x64\php-7.1.5-src\x64\Release_TS\ext\pthreads\ /c ext\pthreads\php_pthreads.c
php_pthreads.c
ext\pthreads\src/pthreads.h(41): fatal error C1083: Cannot open include file: 'netdb.h': No such file or directory
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\cl.exe"' : return code '0x2'
Stop.

I have Windows SDK installed, but that looks like UNIX/POSIX header that is missing

Please help
Momchil

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -2,29 +2,45 @@ version: '{branch}.{build}'
 branches:
   only:
   - master
+
+image: Visual Studio 2017
+
+environment:
+  PHP_SDK_BINARY_TOOLS_VER: php-sdk-2.0.7
+  ARCH: x86 # maybe add x64?
+
+  matrix:
+    - PHP_VER: 7.2
+      VC_VER: vc15
 install:
 - cmd: cinst wget
 build_script:
 - cmd: >-
-    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\vcvars32.bat"
+    "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvars32.bat"
 
-    wget http://windows.php.net/downloads/php-sdk/php-sdk-binary-tools-20110915.zip
+    wget https://github.com/OSTC/php-sdk-binary-tools/archive/%PHP_SDK_BINARY_TOOLS_VER%.zip --no-check-certificate -q -O php-sdk-binary-tools-%PHP_SDK_BINARY_TOOLS_VER%.zip
 
-    7z x -y php-sdk-binary-tools-20110915.zip -oC:\projects\php-sdk
+    7z x -y php-sdk-binary-tools-%PHP_SDK_BINARY_TOOLS_VER%.zip -oC:\projects
 
+    move C:\projects\php-sdk-binary-tools-%PHP_SDK_BINARY_TOOLS_VER% C:\projects\php-sdk
+    
     C:\projects\php-sdk\bin\phpsdk_setvars.bat
 
-    git clone https://github.com/php/php-src C:\projects\php-src
+    git clone https://github.com/php/php-src C:\projects\php-src -b PHP-%PHP_VER% --depth=1
 
     mkdir C:\projects\php-src\ext\pthreads
 
-    xcopy C:\projects\pthreads C:\projects\php-src\ext\pthreads /s /e /y
+    xcopy C:\projects\pthreads C:\projects\php-src\ext\pthreads /s /e /y /q
 
-    wget http://windows.php.net/downloads/php-sdk/deps-7.0-vc14-x86.7z
+    REM FIXME no PHP 7.2 deps downloads available yet - wget http://windows.php.net/downloads/php-sdk/deps-%PHP_VER%-%VC_VER%-%ARCH%.7z -q
+    
+    wget http://windows.php.net/downloads/php-sdk/deps-master-%VC_VER%-%ARCH%.7z -q
 
-    7z x -y deps-7.0-vc14-x86.7z -oC:\projects\php-src
+    REM FIXME no PHP 7.2 deps downloads available yet - 7z x -y deps-%PHP_VER%-%VC_VER%-%ARCH%.7z -oC:\projects\php-src
+    
+    7z x -y deps-master-%VC_VER%-%ARCH%.7z -oC:\projects\php-src
 
-    wget http://www.mirrorservice.org/sites/sources.redhat.com/pub/pthreads-win32/pthreads-w32-2-9-1-release.zip
+    wget http://www.mirrorservice.org/sites/sources.redhat.com/pub/pthreads-win32/pthreads-w32-2-9-1-release.zip -q
 
     7z x -y pthreads-w32-2-9-1-release.zip -oC:\projects\pthreads-w32
 
@@ -34,7 +50,7 @@ build_script:
 
     copy C:\projects\pthreads-w32\Pre-built.2\include\semaphore.h C:\projects\php-src\deps\include\semaphore.h
 
-    copy C:\projects\pthreads-w32\Pre-built.2\lib\x86\pthreadVC2.lib C:\projects\php-src\deps\lib\pthreadVC2.lib
+    copy C:\projects\pthreads-w32\Pre-built.2\lib\%ARCH%\pthreadVC2.lib C:\projects\php-src\deps\lib\pthreadVC2.lib
 
     cd C:\projects\php-src
 
@@ -52,7 +68,7 @@ build_script:
 
     echo extension=php_pthreads.dll > C:\projects\pthreads\bin\modules.d\php.ini
 
-    copy C:\projects\pthreads-w32\Pre-built.2\dll\x86\pthreadVC2.dll C:\projects\pthreads\bin\pthreadVC2.dll
+    copy C:\projects\pthreads-w32\Pre-built.2\dll\%ARCH%\pthreadVC2.dll C:\projects\pthreads\bin\pthreadVC2.dll
 test_script:
 - cmd: C:\projects\pthreads\bin\php.exe -doutput_buffering=0 run-tests.php ext\pthreads -p C:\projects\pthreads\bin\php.exe -q --show-diff
 artifacts:

--- a/src/pthreads.h
+++ b/src/pthreads.h
@@ -38,13 +38,11 @@
 #include <php_globals.h>
 #include <php_main.h>
 #include <php_network.h>
+#ifndef _WIN32
 #include <netdb.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-#ifdef AF_UNIX
 #include <sys/un.h>
-#endif
-#ifndef _WIN32
 #	ifdef HAVE_IF_NAMETOINDEX
 #		include <net/if.h>
 #	endif

--- a/src/socket.c
+++ b/src/socket.c
@@ -206,7 +206,7 @@ void pthreads_socket_bind(zval *object, zend_string *address, zend_long port, zv
 	PTHREADS_SOCKET_CHECK(threaded->store.sock);
 
 	switch (threaded->store.sock->domain) {
-#ifdef AF_UNIX
+#ifndef _WIN32
 		case AF_UNIX: {
 			struct sockaddr_un *sa = (struct sockaddr_un *) sock_type;
 
@@ -357,7 +357,7 @@ void pthreads_socket_connect(zval *object, zend_string *address, zend_long port,
 			RETURN_TRUE;
 		} break;
 
-#ifdef AF_UNIX
+#ifndef _WIN32
 		case AF_UNIX: {
 			struct sockaddr_un s_un = {0};
 
@@ -506,12 +506,13 @@ void pthreads_socket_get_sockaddr(zval *object, zend_long port, struct sockaddr 
 				add_assoc_long(return_value, "port", htons(sin->sin_port));
 			}
 		} break;
-
+#ifndef _WIN32
 		case AF_UNIX: {
 			struct sockaddr_un  *s_un = (struct sockaddr_un *) sa;
 
 			add_assoc_string(return_value, "host", s_un->sun_path);
 		} break;
+#endif
 	}
 }
 

